internal/utils/flags/flags.go

```diff
@@ -243,3 +281,56 @@ func (f StringSliceFlag) Register(fs *pflag.FlagSet) {
 func MarkHidden(fs *pflag.FlagSet, name string) {
 	fs.MarkHidden(name) //nolint: errcheck
 }
+
+var (
+	forwardedNames = make(map[string]string)
+)
+
+// DeprecationHandler is a pflag normalizing function meant to handle deprecating and forwarding flags to new names
+func DeprecationHandler(f *pflag.FlagSet, name string) pflag.NormalizedName {
+	newName, ok := forwardedNames[name]
+	if ok {
+		name = newName
+		f.Lookup(name).Name = newName
```

Comment:
I'm surprised to see this is necessary... i would  have thought some simple implementation was all we needed:

```go
newName, ok := forwardedNames[name]
if ok {
  return pflag.NormalizedName(newName)
}
return pflag.NormalizedName(name)
```

i imagine im missing something here?

Reply:
no need to do it here as well. Sorry these are some experiments I did.